XCUITests/TabsPerformanceTests.swift

```diff
@@ -0,0 +1,137 @@
+                XCTMemoryMetric(),
+                XCTClockMetric(), // to measure timeClock Mon
+                XCTCPUMetric(), // to measure cpu cycles
+                XCTStorageMetric(), // to measure storage consuming
+                XCTMemoryMetric()]) {
+                // activity measurement here
+                app.launch()
+            }
+        }
+    }
+
+    func testPerfTabs20tabTray() {
+        if #available(iOS 13.0, *) {
+            app.launch()
+            waitForExistence(app.buttons["urlBar-cancel"], timeout: 10)
+            navigator.performAction(Action.CloseURLBarOpen)
+            waitForTabsButton()
+            measure(metrics: [
+                XCTClockMetric(), // to measure timeClock Mon
+                XCTCPUMetric(), // to measure cpu cycles
+                XCTStorageMetric(), // to measure storage consuming
+                XCTMemoryMetric()]) {
+                // go to tab tray
+                app.buttons["TabToolbar.tabsButton"].tap()
+                app.buttons["doneButtonTabTray"].tap()
+            }
+        }
+    }
+
+    func testPerfTabs1280tabTray() {
+        if #available(iOS 13.0, *) {
+            app.launch()
+            navigator.performAction(Action.CloseURLBarOpen)
+            waitForTabsButton()
+            measure(metrics: [
+                XCTClockMetric(), // to measure timeClock Mon
+                XCTCPUMetric(), // to measure cpu cycles
+                XCTStorageMetric(), // to measure storage consuming
+                XCTMemoryMetric()]) {
+                // go to tab tray
+                app.buttons["TabToolbar.tabsButton"].tap()
+                app.buttons["doneButtonTabTray"].tap()
+            }
+        }
+    }
+
+    // Additional Scenarios
+    // 1. Start app
+    // 2. Open tabs Tray
+    // 3. Open 1 new tab
+    // 4. Close 1 tab
+    // 5. Cloase all tabs
+    // 6. Switch to private browsing
+    // 7. Switch back from private browsing
+    // 8. Create new tab in private browsing
+
+    // This can be used for generating new tabs
+    /*
+    func testTabs5Setup() {
+        let archiveSize = 1080
+        let fileName = "/Users/rpappalax/git/firefox-ios-TABS-PERF-WIP/Client/Assets/topdomains.txt"
+        var contents = ""
+        var urls = [String]()
+        do {
+            contents = try String(contentsOfFile: fileName)
+            urls = contents.components(separatedBy: "\n")
+        } catch {
+            print("COULDNT LOAD")
+        }
+        navigator.goto(NewTabScreen)
+        var counter = 0
+        let topDomainsCount = urls.count
+        //for url in urls {
+        for n in 0...archiveSize {
+            print(urls[counter])
+            if (counter >= archiveSize) {
+                break
+            }
+            // if we don't have enough urls, start from the beginning of list
+            if (counter >= (topDomainsCount - 1)) {
+                counter = 0
+            }
+            counter += 1
+        }
+    }*/
```

Comment:
If we don't think we'll ever use this scenario, probably time to just clear it 🤷🏼

Reply:
I think this was there as a tool for devs or us to run locally as a custom way to generate as many tabs as you wish